File: MovieMate/frontend/src/app/models/ticket.ts
import { Seat } from "./seat";
import { Showtime } from "./showtime";
import { User } from "./user";

export interface Ticket {
    id?: number;
    orderId?: number;
    user: User;
    showtime: Showtime;
    adultCount: number;
    childCount: number;
    paidAmount: number;
    selectedDate: string;
    orderDate: string;
    isExpanded: boolean;
    assignedSeats: Seat[];
}

File: MovieMate/backend/src/main/java/com/moviemate/entities/Ticket.java
package com.moviemate.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "tickets")
public class Ticket {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long ticketId;

    @ManyToOne
    private User user;

    @ManyToOne
    private Showtime showtime;

    private Long adultCount;
    private Long childCount;
    private Double paidAmount;
    private String selectedDate;

    @PrePersist
    public void generateContractNumber() {
        this.ticketId = generateTicketNumber();
    }

    private Long generateTicketNumber() {
        return Long.valueOf(System.currentTimeMillis() / 5000);
    }
}
